### ISSUE TITLE
Enable Console Commands

### ISSUE THREAD

**Issue by netleader** [mkloubert/vs-deploy#111]
It would be cool if the extension could also execute console commands like: dotnet publish etc.

**Comment by mkloubert** [comment]
@netleader 

Do you mean something like [that](https://github.com/mkloubert/vs-deploy/wiki/targetoperations#open-)?

**Comment by netleader** [comment]
Yes something like that but it's not working.

This is what I've tried. And the console output is just:

`[BEFORE DEPLOY #1] 'open' Opening 'Release'... [OK]`

But the deploy works just not the publish command.
```
"targets": [
            {
                "type": "sftp",
                "name": "Mars",
                "description": "A SFTP folder",
                "host": "hostname", "port": 22,
                "user": "user", "password": "password",
                "dir": "/var/www/dotnet",
                "mappings": [
                    {
                        "source": "MyProject.Linux/bin/Release/netcoreapp2.0/publish",
                        "target": "/"
                    }
                ],
                "beforeDeploy": [
                    {
                        "type": "open",
                        "runInTerminal": false,
                        "target": "dotnet",
                        "arguments": [
                            "publish",
                            "${workspaceRoot}/MyProject.Linux/Bla.Linux.csproj",
                            "-c", "Release"
                        ]
                    }
                ]
            }
        ]
```

**Comment by mkloubert** [comment]
@netleader 

**I have released a new version 9.34.1.**

You should change your `beforeDeploy` settings this way by setting `usePlaceholdersInArguments` to `(true)`:

```json
"beforeDeploy": [
    {
        "type": "open",
        "runInTerminal": false,
        "target": "dotnet",
        "arguments": [
            "publish",
            "${workspaceRoot}/MyProject.Linux/Bla.Linux.csproj",
            "-c", "Release"
        ],

        "usePlaceholdersInArguments": true
    }
]
```

The problem was that the extension could not decide if `${workspaceRoot}` has to be parsed or not, or in other words: if it is an expression for the app you would like to open or a [placeholder](https://github.com/mkloubert/vs-deploy/wiki/values) of the extension.

So you have to setting this behavior explicitly, by using `usePlaceholdersInArguments`.

**Comment by netleader** [comment]
Very nice this works now! Maybe you can have a look into runInTerminal this wont work but not a big thing!

Thank you
